sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQueryBuilder.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using Azure.DigitalTwins.Core.QueryBuilder;
+
+namespace Azure.DigitalTwins.Core.Queries.QueryBuilders
+{
+    /// <summary>
+    /// Used to instantiate an ADT query.
+    /// </summary>
+    public class AdtQueryBuilder
+    {
+        private readonly SelectQuery _selectQuery;
+        private readonly FromQuery _fromQuery;
+        private readonly WhereQuery _whereQuery;
+
+        /// <summary>
+        /// Initializes the clauses that make up a standard query.
+        /// </summary>
+        public AdtQueryBuilder()
+        {
+            _whereQuery = new WhereQuery(this);
+            _fromQuery = new FromQuery(this, _whereQuery);
+            _selectQuery = new SelectQuery(this, _fromQuery);
+        }
+
+        /// <summary>
+        /// Used to select one or multiple collections or properties.
+        /// </summary>
+        /// <param name="args"> The arguments that can be queried (eg *, somePropertyName, etc.) </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            return _selectQuery.Select(args);
+        }
+
+        /// <summary>
+        /// Used when applying the TOP() aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), i.e the number of instances to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (eg property name). </param>
```

Comment:
nit: Instead of `property name` I would use `column name`

Reply:
To be fair, we aren't actually querying table, we're querying a graph, so there technically aren't any columns that we're selecting. I don't mind changing it either way though.